cmd/nitro/nitro.go

```diff
@@ -383,6 +365,33 @@ func mainImpl() int {
 	}
 
 	liveNodeConfig := NewLiveNodeConfig(args, nodeConfig, stackConf.ResolvePath)
+
+	var rollupAddrs chaininfo.RollupAddresses
+	var l1Client *ethclient.Client
+	if nodeConfig.Node.L1Reader.Enable {
+		confFetcher := func() *rpcclient.ClientConfig { return &liveNodeConfig.get().L1.Connection }
+		rpcClient := rpcclient.NewRpcClient(confFetcher, nil)
+		err := rpcClient.Start(ctx)
+		if err != nil {
+			log.Crit("couldn't connect to L1", "err", err)
+		}
+		l1Client = ethclient.NewClient(rpcClient)
+		l1ChainId, err := l1Client.ChainID(ctx)
+		if err != nil {
+			log.Crit("couldn't read L1 chainid", "err", err)
+		}
+		if l1ChainId.Uint64() != nodeConfig.L1.ChainID {
```

Comment:
In the current scheme, `nodeConfig.L1.ChainID` is filled in if missing. As-is I think this requires the user to specify `--l1.chain-id`, which they currently don't need to. Perhaps a better scheme would be to get rid of this field, and instead check that the RPC's L1 chain id matches the L2 chain information we've found (we need to query that information for the rollup addresses anyways).